include/hybridmap/HybridNeutralAtomMapper.hpp

```diff
@@ -0,0 +1,534 @@
+
+  /**
+   * @brief Maps the given quantum circuit to the given architecture and
+   * converts it to the AOD level. Used for Python bindings.
+   * @param qc  The quantum circuit to be mapped
+   * @param initialMapping The initial mapping of the circuit qubits to the
+   * hardware qubits
+   */
+  [[maybe_unused]] void mapPy(qc::QuantumComputation& qc,
+                              InitialMapping initialMapping, bool printInfo) {
+    map(qc, initialMapping, verbose = printInfo);
+    convertToAod(this->mappedQc);
+  }
+
+  /**
+   * @brief Prints the mapped circuits as an extended OpenQASM string.
+   * @return The mapped quantum circuit with abstract SWAP gates and MOVE
+   */
+  [[maybe_unused]] std::string getMappedQc() {
+    std::stringstream ss;
+    this->mappedQc.dumpOpenQASM(ss, false);
+    return ss.str();
+  }
+
+  /**
+   * @brief Saves the mapped quantum circuit to a file.
+   * @param filename The name of the file to save the mapped quantum circuit to
+   */
+  [[maybe_unused]] void saveMappedQc(const std::string& filename) {
+    std::ofstream ofs(filename);
+    this->mappedQc.dumpOpenQASM(ofs, false);
+  }
+
+  /**
+   * @brief Prints the mapped circuit with AOD operations as an extended
+   * OpenQASM
+   * @return The mapped quantum circuit with native AOD operations
+   */
+  [[maybe_unused]] std::string getMappedQcAOD() {
+    std::stringstream ss;
+    this->mappedQcAOD.dumpOpenQASM(ss, false);
+    return ss.str();
+  }
+
+  /**
+   * @brief Saves the mapped quantum circuit with AOD operations to a file.
+   * @param filename The name of the file to save the mapped quantum circuit
+   * with AOD operations to
+   */
+  [[maybe_unused]] void saveMappedQcAOD(const std::string& filename) {
+    std::ofstream ofs(filename);
+    this->mappedQcAOD.dumpOpenQASM(ofs, false);
+  }
+
+  /**
+   * @brief Schedules the mapped quantum circuit on the neutral atom
+   * architecture.
+   * @details For each gate/operation in the input circuit, the scheduler checks
+   * the earliest possible time slot for execution. If the gate is a multi qubit
+   * gate, also the blocking of other qubits is taken into consideration. The
+   * execution times are read from the neutral atom architecture.
+   * @param verboseArg If true, prints additional information
+   * @param createAnimationCsv If true, creates a csv file for the animation
+   * @param shuttlingSpeedFactor The factor to speed up the shuttling time
+   * @return The results of the scheduler
+   */
+  [[maybe_unused]] SchedulerResults schedule(bool verboseArg           = false,
+                                             bool createAnimationCsv   = false,
+                                             fp   shuttlingSpeedFactor = 1.0) {
+    scheduler = NeutralAtomScheduler(arch);
+    return scheduler.schedule(mappedQcAOD, hardwareQubits.getInitialHwPos(),
+                              verboseArg, createAnimationCsv,
+                              shuttlingSpeedFactor);
+  }
```

Comment:
Should scheduling really be part of the mapper here?
wouldn't it make sense to decouple this and have a mapping stage that is followed by a separate scheduling stage?

Reply:
I have the two stages sperate from each other. This is just a shortcut if one wants to do (as commonly done) both and for the Python bindings.

Reply:
Fair enough. Maybe I would have envisioned more of a `NeutralAtomCompiler` meta class that orchestrates Synthesis, Mapping, Scheduling, Conversion, etc. instead of having this "mangled" in the Neutral Atom Mapper. I guess it's not really necessary to change anything here right now, but It could be helpful in the future to have some top level orchestration class that manages everything.

Just as context: we have something like that in QCEC, where there is an `EquivalenceCheckingManager` that orchestrates the whole flow and the individual and isolated `EquivalenceChecker`s that are being managed. Makes for a very clean separation of concerns.

Reply:
Definitely would make sense. I think the reason to it this way was also because I cannot convert the mapped circuit back to Python (Qiskit) as it is no longer valid qasm. Once we manage to unify this using e.g. Opaque gates, this should be no problem.

Reply:
Yeah, I see where you are coming from.
Let's keep this in mind when we discuss the general unification for the compiler collection. Feel free to resolve for now.